lib/internal/errors.js

```diff
@@ -59,12 +59,17 @@ const {
   URIError,
 } = primordials;
 
+const {
+  assignOwnProperties,
+  setOwnProperty,
+} = require('internal/util/safe-property-assignment');
+
 const kIsNodeError = Symbol('kIsNodeError');
 
 const isWindows = process.platform === 'win32';
 
 const messages = new SafeMap();
-const codes = {};
+const codes = ObjectCreate(null);
```

Comment:
```suggestion
const codes = { __proto__: null };
```
this is equally robust and avoids a function call.

Reply:
I do prefer this, but I know Antoine prefers the other, and the diff is pretty "6 of one".

Reply:
Why the preference?

Even setting aside that Object.create is a regretted addition to the JS language, it's still got the potential overhead of a function call, as compared to the very straightforward syntax that node uses on objects initialized with more than one property. Why is "no properties" a special case that requires a function call?

Reply:
> it's still got the potential overhead of a function call

I don't think that should be a concern here because `lib/internal/errors.js` is already a part of the snapshot, so this code won't get executed at startup, rather the context would get deserialized from the snapshot.

Reply:
ok, fair - it’s still indirection, and it’s using an API call for something that’s available with syntax.

Reply:
This is a style question that’s out of scope for this PR. If someone feels strongly about this, they should open a new PR that updates our eslint to enforce a particular style.

Reply:
opened here: #46083

Reply:
(that was landed, so this can be resolved)